gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -962,4 +967,915 @@ public void testListDnsRecords() {
+          Dns.RecordSetListOption.dnsName(A_RECORD_ZONE1.name()));
+      batch.submit();
+      recordSetIterator = result.get().iterateAll();
+      counter = 0;
+      while (recordSetIterator.hasNext()) {
+        RecordSet recordSet = recordSetIterator.next();
+        assertTrue(ImmutableList.of(A_RECORD_ZONE1.type(), AAAA_RECORD_ZONE1.type())
+            .contains(recordSet.type()));
+        counter++;
+      }
+      assertEquals(2, counter);
+      // test type filter
+      batch = DNS.batch();
+      result = batch.listRecordSets(ZONE1.name(),
+          Dns.RecordSetListOption.dnsName(A_RECORD_ZONE1.name()),
+          Dns.RecordSetListOption.type(A_RECORD_ZONE1.type()));
+      batch.submit();
+      recordSetIterator = result.get().iterateAll();
+      counter = 0;
+      while (recordSetIterator.hasNext()) {
+        RecordSet recordSet = recordSetIterator.next();
+        assertEquals(A_RECORD_ZONE1, recordSet);
+        counter++;
+      }
+      assertEquals(1, counter);
+      batch = DNS.batch();
+      DnsBatchResult<Page<RecordSet>> noNameError = batch.listRecordSets(ZONE1.name(),
+          Dns.RecordSetListOption.type(A_RECORD_ZONE1.type()));
+      DnsBatchResult<Page<RecordSet>> zeroSizeError =
+          batch.listRecordSets(ZONE1.name(), Dns.RecordSetListOption.pageSize(0));
+      DnsBatchResult<Page<RecordSet>> negativeSizeError =
+          batch.listRecordSets(ZONE1.name(), Dns.RecordSetListOption.pageSize(-1));
+      batch.submit();
+      // check wrong arguments
+      try {
+        // name is not set
+        noNameError.get();
+        fail();
+      } catch (DnsException ex) {
+        // expected
+        assertEquals(400, ex.code());
+        assertFalse(ex.retryable());
+      }
+      try {
+        zeroSizeError.get();
+        fail();
+      } catch (DnsException ex) {
+        // expected
+        assertEquals(400, ex.code());
+        assertFalse(ex.retryable());
+      }
+      try {
+        negativeSizeError.get();
+        fail();
+      } catch (DnsException ex) {
+        // expected
+        assertEquals(400, ex.code());
+        assertFalse(ex.retryable());
+      }
+      waitForChangeToComplete(ZONE1.name(), change.generatedId());
+    } finally {
+      clear();
+    }
+  }
+
+  @Test
+  public void testBatchCombined() {
+    // only testing that the combination is possible
+    // the results are validated in the other test methods
+    try {
+      DNS.create(ZONE1);
+      DnsBatch batch = DNS.batch();
+      DnsBatchResult<Zone> zoneResult = batch.getZone(ZONE_NAME1);
+      DnsBatchResult<ChangeRequest> changeRequestResult = batch.getChangeRequest(ZONE_NAME1, "0");
+      DnsBatchResult<Page<RecordSet>> pageResult = batch.listRecordSets(ZONE_NAME1);
+      DnsBatchResult<ProjectInfo> projectResult = batch.getProject();
+      assertFalse(zoneResult.completed());
+      try {
+        zoneResult.get();
+        fail("this should be submitted first");
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2428842112/issues/source?bid=3123103&fileBranchId=3240468#l1863)